skbio/metadata/_interval.py

```diff
@@ -512,6 +520,30 @@ def __init__(self, upper_bound):
         # Indicates if the IntervalTree needs to be rebuilt.
         self._is_stale_tree = False
 
+        if copy_from is not None:
+            for interval in copy_from._intervals:
+                # Only need to shallow-copy `bounds` and `fuzzy`
+                # because their elements are immutable.
+                bounds_cp = interval.bounds[:]
+                fuzzy_cp = interval.fuzzy[:]
+                metadata_cp = copy.copy(interval.metadata)
+
+                self.add(bounds_cp, fuzzy=fuzzy_cp, metadata=metadata_cp)
+
+    def _rebuild_tree(method):
```

Comment:
You are talking about renaming from `self` to `interval_metadata`? That's the only change I made. It can stay it was. But I feel like it makes more sense with the new name. BTW, I think this function better live outside the class? Just to be clear, it can't be `@staticmethod` decorated.